src/Worlds.js

```diff
@@ -26,58 +36,95 @@ import { ReactComponent as SpaceContrast } from './svg/Space-contrast.svg';
-
-// Characters
-import { ReactComponent as RabbitIcon } from './svg/Rabbit.svg';
-import { ReactComponent as RobotIcon } from './svg/Robot.svg';
-import { ReactComponent as SpaceShipIcon } from './svg/SpaceShip.svg';
+import { ReactComponent as SpaceShip } from './svg/SpaceShip.svg';
+import { ReactComponent as SpaceShipGray } from './svg/SpaceShip-gray.svg';
+import { ReactComponent as SpaceShipContrast } from './svg/SpaceShip-contrast.svg';
 
 export type WorldProperties = {
-    background: any,
-    backgroundGray: any,
-    backgroundContrast: any,
-    thumbnail: any,
-    thumbnailGray: any,
-    thumbnailContrast: any,
-    character: any
+    background: ?React.ComponentType<{}>,
+    backgroundGray: ?React.ComponentType<{}>,
+    backgroundContrast: ?React.ComponentType<{}>,
+    thumbnail: React.ComponentType<{}>,
+    thumbnailGray: React.ComponentType<{}>,
+    thumbnailContrast: React.ComponentType<{}>,
+    character: React.ComponentType<{}>
 };
 
 const worlds = {
     'DeepOcean': {
         background: DeepOcean,
         backgroundGray: DeepOceanGray,
         backgroundContrast: DeepOceanContrast,
+        backgroundInfo: [
+            [null, null, null, null, null, null, null, null, null, null, null, null],
+            [null, null, null, null, null, null, null, null, null, 'DeepOcean.fishGroup', 'DeepOcean.fishGroup', null],
+            [null, null, null, null, null, null, null, null, null, 'DeepOcean.fishGroup', 'DeepOcean.fishGroup', null],
+            [null, 'DeepOcean.babyJellyfish', 'DeepOcean.jellyfish', null, null, null, 'DeepOcean.shark', 'DeepOcean.shark', null, null, 'DeepOcean.shark', 'DeepOcean.coral'],
+            [null, 'DeepOcean.babyJellyfish', 'DeepOcean.jellyfish', null, 'DeepOcean.shark', 'DeepOcean.shark', 'DeepOcean.shark', 'DeepOcean.shark', 'DeepOcean.shark', 'DeepOcean.shark', 'DeepOcean.shark', 'DeepOcean.coral'],
+            [null, null, null, null, 'DeepOcean.shark', 'DeepOcean.shark', 'DeepOcean.shark', 'DeepOcean.shark', null, null, null, null],
+            ['DeepOcean.fish', 'DeepOcean.coral', null, null, null, null, null, 'DeepOcean.treasure', 'DeepOcean.treasure', 'DeepOcean.treasure', null, null],
+            ['DeepOcean.coral', 'DeepOcean.fish', null, null, null, null, null, 'DeepOcean.treasure', 'DeepOcean.treasure', 'DeepOcean.treasure', null, 'DeepOcean.fish']
+        ],
         thumbnail: DeepOceanThumbnail,
         thumbnailGray: DeepOceanThumbnailGray,
         thumbnailContrast: DeepOceanThumbnailContrast,
-        character: RobotIcon
+        character: Submarine,
+        characterGray: SubmarineGray,
+        characterContrast: SubmarineContrast
     },
     'Jungle': {
         background: Jungle,
         backgroundGray: JungleGray,
         backgroundContrast: JungleContrast,
+        backgroundInfo: [
+            [null, null, null, null, null, null, 'Jungle.giraffe', null, null, 'Jungle.lion', null, null],
```

Comment:
I also think we could do more with type checking here, whether or not we change the structure.  One approach would be to have a type for the items in each group, and to OR those together if we want a set of all supported items.

I think it's probably a good idea to decouple the item name from message keys as well, as we might want to do things other than just announce them.  So, you might choose to announce `"{World}.{cellContents}"`, but I might want to check `cellContents` separately and play a sound or animation for "giraffe" or "lion".  It makes more sense to keep the message key scoping out in that case.

It also makes it easier to share objects between worlds, for example, if we add goals or other things that are the same in multiple worlds.